CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -3,32 +3,31 @@
 #include <iostream>
 #include <pthread.h>
 #include <unistd.h>
+#include <vector>
 
 std::ifstream CallBack::_outFile;
 
 void* CallBack::OutputThread(void*)
 {
-  if (_outFile)
-  {
+  if (_outFile.is_open()) {
     int pos = 0;
-    while (!_outFile.eof()) 
+    std::vector<char> buffer(size_t(4096));
+    while (_outFile.good() && !_outFile.eof()) 
     {
       _outFile.seekg(0, _outFile.end);
       int length = static_cast<int>(_outFile.tellg()) - pos;
       _outFile.seekg(pos, _outFile.beg);
  
       if (length > 0) 
       {
-        char* buffer = new char[length + 1];
-        _outFile.read(buffer, length);
-        buffer[length] = '\0';
+        _outFile.read(buffer.data(), length);
         pos += length;
-        std::cout << buffer;
-        delete[] buffer;
+        std::cout.write(buffer.data(), length);
       }
+      _outFile.clear();
```

Comment:
@JoshDreamland so apparently this line is necessary for it to work on Windows and also the reason why AppVeyor wasn't printing the full log correctly. Can you please carefully review this function and the changes to make sure that it works completely as I expect it to and I don't have to change it again.